app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,26 @@
+class SessionsController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.find_by(name: params[:user][:name])
+    if @user
+      session[:user_id] = @user.id
+      flash[:success] = "Successfully logged in as existing user #{@user.name}."
+    else
+      @user = User.create(name: params[:user][:name])
```

Comment:
You should have a check here to see if the `create` worked.  What if someone tried to login with a space in their name?